### PR TITLE
Apigateway widgets

### PR DESCRIPTION
**Pull request by anetteOlli**

### CODE REVIEW

**Comment by anetteOlli** [comment]
Lagt til Apigateway monitorering:

image

File: dashboard/src/main/java/no/sikt/nva/monitoring/model/CloudWatchWidget.java [hunk 1]
```diff
@@ -1,7 +1,27 @@
 package no.sikt.nva.monitoring.model;
 
+import java.util.Objects;
 import no.unit.nva.commons.json.JsonSerializable;
+import nva.commons.core.JacocoGenerated;
 
 public record CloudWatchWidget<I> (String type, I properties, int height, int width, int x, int y) implements JsonSerializable {
 
+    @JacocoGenerated
+    @Override
```
Comment (anetteOlli) [inline dashboard/src/main/java/no/sikt/nva/monitoring/model/CloudWatchWidget.java:10]: override the equals method, as I don't think it is vital for checking the x, y, width and height properties.

File: dashboard/src/test/java/no/sikt/nva/monitoring/UpdateDashboardHandlerTest.java [hunk 2]
```diff
@@ -28,25 +35,63 @@
...
+        TYPE, METRIC_PROPERTIES_COUNT, IGNORED, IGNORED, IGNORED, IGNORED).toJsonString();
     private FakeCloudWatchClient cloudWatchClient;
     private UpdateDashboardHandler handler;
+    private FakeApiGatewayClient apiGatewayClient;
 
     @BeforeEach
     void init() {
         cloudWatchClient = new FakeCloudWatchClient();
-        handler = new UpdateDashboardHandler(cloudWatchClient);
+        apiGatewayClient = new FakeApiGatewayClient();
```
Comment (torbjokv) [inline dashboard/src/test/java/no/sikt/nva/monitoring/UpdateDashboardHandlerTest.java:88]: I would personally prefer mock(ApiGatewayClient.class) but whatever works :-)

File: template.yml [hunk 3]
```diff
@@ -64,13 +64,28 @@ Resources:
...
+      PolicyName: !Sub ApiGatewayPermissions-${AWS::StackName}
+      Roles:
+        - !Ref LambdaRole
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Action:
+              - Apigateway:GET
+            Resource: "*"
```
Comment (torbjokv) [inline template.yml:79]: not sure what endpoint this is needed for, but could it be that you could limit it to  'arn:aws:apigateway:*::/restapis' without any wildcard? as far as I understand, this give get access to all our apis (I think?) If you also need 'arn:aws:apigateway:<region>::/restapis/api-id-1, 2,3....' then I guess it's more tricky

File: dashboard/src/main/java/no/sikt/nva/monitoring/model/CloudWatchWidget.java (hunk 1 above)
Comment (torbjokv) [reply dashboard/src/main/java/no/sikt/nva/monitoring/model/CloudWatchWidget.java:10]: If we want to update the layout, would this cause it to not update?

**Review by torbjokv — CHANGES_REQUESTED** [review]

File: dashboard/src/main/java/no/sikt/nva/monitoring/model/CloudWatchWidget.java (hunk 1 above)
Comment (anetteOlli) [reply dashboard/src/main/java/no/sikt/nva/monitoring/model/CloudWatchWidget.java:10]: Nope, it is only when asserting that the widgets are created during tests.

**Review by brinxmat — COMMENTED** [review]
Not much to add, but I agree that `Resource: *` seems a bit liberal

File: template.yml (hunk 3 above)
Comment (torbjokv) [reply template.yml:79]: probably ok as we need invoke to actually invoke

**Review by torbjokv — APPROVED** [review]

**Review by brinxmat — APPROVED** [review]

**Pull request merged**